privacy/i2p.md

```diff
@@ -0,0 +1,317 @@
+---
+layout: doc
+title: transparent-i2p-proxyvm
+permalink: /doc/i2p/
+---
+
+# I2P #
+The Invisible Internet Project[I2P] is an anonymising peer-to-peer network that enables access to hidden I2P services similar to [Tor Hidden Services][hidden-servies].
+Though there are outproxies (to access the clearnet), they are rare and the I2P-network isn't built to be used as anonymising network, but much rather an anonymity-by-default secondary network.
+
+# Transparent ProxyVM #
+The currently available I2P-software is built to provide a simple http-proxy to route trafic through the network.
+This creates problems with software that can't handle proxies or handles them badly.
+A transparent proxy allows us to take all trafic and funnel it through the I2P-network to prevent information leakage.  
+Furthermore, a transparent proxy allows us to handle traffic to the I2P-network while also handling tor-traffic.
+
+# Theory of operation #
+DNS requests from any appVM routed through the I2P-proxyVM are intercepted and answered with a localnet-IP if the domain ends in `.i2p`, otherwise the tor-dns-service is requested.
+Any http requests to the localnet-IP again is intercepted and routed to tinyproxy, which repackages the requests into proxy-requests and sends them to the local I2P-server.
+All other tcp-requests are routed through TOR.
+
+# Tor #
+Torifying traffic allows access to tor hidden nodes.
+Assuming you want to access all kind of hidden services, not just the I2P network, it's a good idea to handle tor traffic parallel to I2P
+
+# DISCLAIMER #
+Assume I have no idea what I'm doing.
+The result of this guide should be seen more adding access to I2P Hidden Services more than protecting your privacy with the I2P network.
+
+## Quick Note on I2P-Over-Tor ##
+There is an excellent guide on how to use [I2P over TOR via whonix][whonix-I2P].
+However, this adds load to the tor-network.
+Since the main purpose of this guide is to access the I2P-network, the performance cost overshadowed the security benefits.
+
+## Qubes 3.2 ##
+### TemplateVM setup ###
+This guide builds a proxyVM based on a Debian template.
+
+* **(Optional)** If you want to use a separate vm template for your I2P-proxyVM, clone your Debian template: (run in dom0)*
+
+    qvm-clone debian-8 debian-8-gate
+
+### Install the necessary tools in the templateVM ###
+I2P install instructions can be found [here][i2p-debian-instructions].
+This guide assumes debian-8, other versions please reference the according instructions.
+TOR can be installed from default debian-repos.
+tinyproxy and dnsmasq can be installed from debian-repos, but are already installed by default in qubes' debian-8.
+
+1. Add the repo-key for I2P's Debian repository to apt:
+  Download:
+
+      wget https://geti2p.net/_static/i2p-debian-repo.key.asc
+
```

Comment:
Use curl for better security.

    curl --tlsv1.2 --proto =https --location --remote-name

https://www.whonix.org/wiki/Secure_Downloads

Reply:
I tried using as little additional installs as possible. wget is preinstalled in debian 8 and 9 templates, curl is not. Since the fingerprint is checked later, is there a big advantage in using curl/scurl?